tests/js/dataMock/analytics.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * Analytics test data mock.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import faker from 'faker';
+import { zip, from, Observable } from 'rxjs';
+import { map, reduce, take } from 'rxjs/operators';
+
+const ANALYTICS_METRIC_TYPES = {
+	'ga:users': 'INTEGER',
+	'ga:newUsers': 'INTEGER',
+	'ga:sessions': 'INTEGER',
+	'ga:goalCompletionsAll': 'INTEGER',
+	'ga:pageviews': 'INTEGER',
+	'ga:uniquePageviews': 'INTEGER',
+	'ga:bounceRate': 'PERCENT',
+	'ga:avgSessionDuration': 'TIME',
+	'ga:adsensePageImpressions': 'INTEGER',
+	'ga:adsenseCTR': 'PERCENT',
+	'ga:adsenseRevenue': 'CURRENCY',
+	'ga:adsenseECPM': 'CURRENCY',
+};
+
+const ANALYTICS_DIMENSION_OPTIONS = {
+	'ga:channelGrouping': [
+		'Organic Search',
+		'Referral',
+		'Direct',
+		'(other)',
+	],
+	'ga:country': [
+		'United States',
+		'United Kingdom',
+		'India',
+		'(not set)',
+		'France',
+		'Ukraine',
+		'Italy',
+		'Mexico',
+	],
+	'ga:deviceCategory': [
+		'desktop',
+		'tablet',
+		'mobile',
+	],
+	'ga:pageTitle': ( i ) => i <= 12 ? `Test Post ${ i }` : false,
+	'ga:pagePath': ( i ) => i <= 12 ? `/test-post-${ i }/` : false,
```

Comment:
To limit the number of fake pages to 12.

Reply:
Right, I guess the limitation there seems both inconsistent and arbitrary. Why not always return the numbered string?

Reply:
Can you show me an example of what you mean? We use `false` to stop generators after the 12th item.

Reply:
A few things which aren't quite clear for me yet are:

- Why specifically `12`?  
`90` is commented below which makes sense as it relates to a date range but 12 seems arbitrary and isn't explained
- Why limit the length of the generated items based on the dimension?

Why not simplify it like so?

```js
const ANALYTICS_DIMENSION_OPTIONS = {
	// ...
	'ga:pageTitle': ( i ) => `Test Post ${ i }`,
	'ga:pagePath': ( i ) => `/test-post-${ i }/`,
};
// ... below
streams.push( new Observable( ( observer ) => {
	for ( let i = 1; i <= 90; i++ ) { // 90 is the max number of dates in the longest date range.
		const val = ANALYTICS_DIMENSION_OPTIONS[ dimension ]( i );
		observer.next( val );
	}
	observer.complete();
} ) );
```

Reply:
See IB:

```
For [ 'ga:pageTitle', 'ga:pagePath' ], rely on a hard-coded list of 12 "web pages". Simply generate them e.g. by a for loop and use "Test Post %d" for the ga:pageTitle and "/test-post-%d/" for the ga:pagePath. Then iterate over these hard-coded values, i.e. the response should include 12 rows.
```

Reply:
Thanks for pointing me to that, but this still doesn't explain why it should be done, and why only 12?

The ACs only say this about dimensions:
> The function should support only the following _specific_ `dimensions` values for now:
> * `[ 'ga:date' ]`
> * `[ 'ga:channelGrouping' ]`
> * `[ 'ga:pageTitle', 'ga:pagePath' ]`

It looks like @felixarntz wrote the IB as well so perhaps we don't know why, but if this were a requirement I would expect it to be mentioned in the ACs 😄

Reply:
I'm still not quite sure why this limitation is in place; it seems to break the expected output. Assigning to @felixarntz to give it a final review since he has the full context about the reasoning here. It doesn't need to come back to me though – if he's good with it then 👍 👍  from me.

Reply:
@aaemnnosttv @eugene-manuilov This limitation was based on my implementation in the tester plugin. We'll need to limit support to specific dimensions since we should be using realistic values. For example, `ga:date` should rely on the appropriate dates on request parameters, `ga:channelGrouping` should use actual dimensions, `ga:pageTitle` should use example page titles. This is the same that applies to metrics, we cannot just let any metric go through because that would require way too much work; so we should focus on those that are actually relevant for Site Kit.